lib/archethic/contracts/interpreter/library.ex

```diff
@@ -217,4 +194,26 @@ defmodule Archethic.Contracts.Interpreter.Library do
       bin
     end
   end
+
+  @doc """
+  Get the genesis address of the chain
+  """
+  @spec get_genesis_address(binary()) ::
+          binary()
+  def get_genesis_address(address) do
+    addr_bin = decode_binary(address)
+    {:ok, address} = Archethic.TransactionChain.fetch_genesis_address_remotely(addr_bin)
+    Base.encode16(address)
+  end
+
+  @doc """
+  Get the First transaction address of the transaction chain for the given address
+  """
+  @spec get_first_transaction_address(address :: binary()) ::
+          binary()
+  def get_first_transaction_address(address) do
+    addr_bin = decode_binary(address)
+    {:ok, address} = Archethic.TransactionChain.fetch_first_transaction_address_remotely(addr_bin)
+    Base.encode16(address)
```

Comment:
I believe you need to deal with `{:ok, :not_exists}`

Reply:
alias TransactionChain also